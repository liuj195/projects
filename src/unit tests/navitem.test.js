import React from "react";
import DropdownItem from "../components/navbar/navitem";
import renderer from "react-test-renderer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

test("Navitems render correctly", () => {
  const component = renderer.create(
    <DropdownItem
      printFile={() => {}}
      onUpload={() => {}}
      onSaveFile={() => {}}
      onLegendClick={() => {}}
      onChangeLayout={() => {}}
      itemList={["testing"]}
      user={{ userID: "abc", access: [] }}
      key={1}
      name="item"
      isDisabled={true}
      styleProps={{ color: "green" }}
      subMenus={{
        Layout: ["test", "abc"],
        Show_Hide: ["testing", "123"]
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  //simulate click on child
  tree[1].children[0].props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // blur on child
  tree[1].children[0].props.onBlur();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  //mouse leave on child
  tree[1].children[1].props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  //mouse enter on grandchild
  tree[1].children[1].children[0].children[0].props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
