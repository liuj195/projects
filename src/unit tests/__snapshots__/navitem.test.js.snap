// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navitems render correctly 1`] = `
Array [
  <input
    id="myInput"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />,
  <li
    style={null}
  >
    <button
      aria-controls="basic-nav-section-one"
      aria-expanded={false}
      className="usa-accordion-button usa-nav-link switchBackground"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      style={
        Object {
          "color": "grey",
        }
      }
    >
      <span>
        item
      </span>
    </button>
    <ul
      aria-hidden={true}
      className="usa-nav-submenu"
      id="basic-nav-section-one"
      onMouseLeave={[Function]}
    >
      <div
        className="testing"
      >
        <li
          className="dropDownItem"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          style={
            Object {
              "color": "white",
              "marginLeft": 4,
            }
          }
        >
          testing
        </li>
      </div>
    </ul>
  </li>,
]
`;

exports[`Navitems render correctly 2`] = `
Array [
  <input
    id="myInput"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />,
  <li
    style={null}
  >
    <button
      aria-controls="basic-nav-section-one"
      aria-expanded={true}
      className="usa-accordion-button usa-nav-link switchBackground"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      style={
        Object {
          "color": "grey",
        }
      }
    >
      <span>
        item
      </span>
    </button>
    <ul
      aria-hidden={false}
      className="usa-nav-submenu"
      id="basic-nav-section-one"
      onMouseLeave={[Function]}
    >
      <div
        className="testing"
      >
        <li
          className="dropDownItem"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          style={
            Object {
              "color": "white",
              "marginLeft": 4,
            }
          }
        >
          testing
        </li>
      </div>
    </ul>
  </li>,
]
`;

exports[`Navitems render correctly 3`] = `
Array [
  <input
    id="myInput"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />,
  <li
    style={null}
  >
    <button
      aria-controls="basic-nav-section-one"
      aria-expanded={false}
      className="usa-accordion-button usa-nav-link switchBackground"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      style={
        Object {
          "color": "grey",
        }
      }
    >
      <span>
        item
      </span>
    </button>
    <ul
      aria-hidden={true}
      className="usa-nav-submenu"
      id="basic-nav-section-one"
      onMouseLeave={[Function]}
    >
      <div
        className="testing"
      >
        <li
          className="dropDownItem"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          style={
            Object {
              "color": "white",
              "marginLeft": 4,
            }
          }
        >
          testing
        </li>
      </div>
    </ul>
  </li>,
]
`;

exports[`Navitems render correctly 4`] = `
Array [
  <input
    id="myInput"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />,
  <li
    style={null}
  >
    <button
      aria-controls="basic-nav-section-one"
      aria-expanded={false}
      className="usa-accordion-button usa-nav-link switchBackground"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      style={
        Object {
          "color": "grey",
        }
      }
    >
      <span>
        item
      </span>
    </button>
    <ul
      aria-hidden={true}
      className="usa-nav-submenu"
      id="basic-nav-section-one"
      onMouseLeave={[Function]}
    >
      <div
        className="testing"
      >
        <li
          className="dropDownItem"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          style={
            Object {
              "color": "white",
              "marginLeft": 4,
            }
          }
        >
          testing
        </li>
      </div>
    </ul>
  </li>,
]
`;

exports[`Navitems render correctly 5`] = `
Array [
  <input
    id="myInput"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />,
  <li
    style={null}
  >
    <button
      aria-controls="basic-nav-section-one"
      aria-expanded={false}
      className="usa-accordion-button usa-nav-link switchBackground"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      style={
        Object {
          "color": "grey",
        }
      }
    >
      <span>
        item
      </span>
    </button>
    <ul
      aria-hidden={true}
      className="usa-nav-submenu"
      id="basic-nav-section-one"
      onMouseLeave={[Function]}
    >
      <div
        className="testing"
      >
        <li
          className="dropDownItem"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          style={
            Object {
              "color": "white",
              "marginLeft": 4,
            }
          }
        >
          testing
        </li>
      </div>
    </ul>
  </li>,
]
`;
